rotkehlchen/globaldb/handler.py

```diff
@@ -1011,6 +1011,108 @@ def add_historical_prices(entries: List['HistoricalPrice']) -> None:
 
         connection.commit()
 
+    @staticmethod
+    def add_single_historical_price(entry: HistoricalPrice) -> None:
+        """Adds the given historical price entries in the DB"""
+        connection = GlobalDBHandler()._conn
+        cursor = connection.cursor()
+        try:
+            serialized = entry.serialize_for_db()
+            cursor.execute(
+                """INSERT OR IGNORE INTO price_history(
+                from_asset, to_asset, source_type, timestamp, price
+                ) VALUES (?, ?, ?, ?, ?)
+                """,
+                serialized,
+            )
+        except sqlite3.IntegrityError as e:
+            connection.rollback()
+            log.error(
+                f'Failed to add single historical price. {str(e)}. ',
+            )
+        # success
+        connection.commit()
+
+    @staticmethod
+    def get_manual_prices(
+        from_asset: Optional[Asset],
+        to_asset: Optional[Asset],
+    ) -> List[Dict[str, str]]:
+        """Returns prices added to the database by the user for an asset"""
+        connection = GlobalDBHandler()._conn
+        cursor = connection.cursor()
+        querystr = (
+            'SELECT from_asset, to_asset, source_type, timestamp, price FROM price_history '
+            'WHERE source_type="A" '
+        )
+        params = []
+        if from_asset is not None:
+            querystr += 'AND from_asset=? '
+            params.append(from_asset.identifier)
+        if to_asset is not None:
+            querystr += 'AND to_asset=? '
+            params.append(to_asset.identifier)
+        querystr += 'ORDER BY timestamp'
+        query = cursor.execute(querystr, tuple(params))
+        result = query.fetchall()
+        return [
+            {
+                'from_asset': entry[0],
+                'to_asset': entry[1],
+                'timestamp': entry[3],
+                'price': entry[4],
+            }
+            for entry in result
+        ]
+
+    @staticmethod
+    def edit_manual_price(entry: HistoricalPrice) -> bool:
+        """Edits a manually insert historical price. Returns false if no row
+        was updated and true otherwise.
+        """
+        connection = GlobalDBHandler()._conn
+        cursor = connection.cursor()
+        querystr = (
+            'UPDATE price_history SET price=? WHERE from_asset=? AND to_asset=? '
+            'AND source_type=? AND timestamp=? '
+        )
+        entry_serialized = entry.serialize_for_db()
+        # Price that is the last entry should be the first and the rest of the
+        # positions are correct in the tuple
+        params_update = entry_serialized[-1:] + entry_serialized[:-1]
+        try:
+            changes = cursor.execute(querystr, params_update).rowcount
```

Comment:
That's fine if it works. But just making sure you also know that you can also get the same thing from the next line by just using the same cursor. So like this:

```python
cursor.execute(querystr, params_update)
changes = cursor.rowcount
```